crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -105,6 +106,99 @@ pub(super) const END_EXPR_SET: TokenSet = TokenSet::new([
 /// Tokens that can appear at the end of a sequence.
 const END_SEQUENCE_SET: TokenSet = END_EXPR_SET.remove(TokenKind::Comma);
 
+/// Represents the expression parsing context.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub(super) struct ExpressionContext(ExpressionContextFlags);
+
+impl Default for ExpressionContext {
+    /// Sets the default flags for the expression context which is to:
+    /// * Consider `in` keyword in a comparison expression
+    /// * Disallow starred and yield expressions
+    fn default() -> Self {
+        ExpressionContext(ExpressionContextFlags::INCLUDE_IN)
+    }
+}
+
+bitflags! {
+    #[derive(Debug, Copy, Clone, PartialEq, Eq)]
+    struct ExpressionContextFlags: u8 {
+        /// This flag is set when the `in` keyword should be included as part of a comparison
+        /// expression. It is to avoid ambiguity in `for ... in ...` statements. This flag is
+        /// set by default.
+        const INCLUDE_IN = 1 << 0;
```

Comment:
Nit: You could change it to `EXCLUDE_IN` so that you can remove the `default` implementation

Nit: I think it should otherwise be `ALLOW_IN` to align with the other flags.

Reply:
I like the `EXCLUDE_IN` which is what we actually want to set. Thanks for the suggestion!